### ISSUE TITLE
Unclear code in prepare_train_labels

### ISSUE THREAD

**Issue by ParikhKadam** [Daniil-Osokin/lightweight-human-pose-estimation.pytorch#40]
I am trying to understand the code.. and I have some queries.

This line https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch/blob/3fced2c8fb771a82e26246691de6a3cd9b4e05b9/scripts/prepare_train_labels.py#L101 specifies that if there's a crowd, annotation for this image won't be added to `annotations_per_image_mapping`.

The line here https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch/blob/3fced2c8fb771a82e26246691de6a3cd9b4e05b9/scripts/prepare_train_labels.py#L108 indicates that the image will only be added to `crowd_segmentations_per_image_mapping` if it has a crowd in it.

Hence, I don't think this condition https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch/blob/3fced2c8fb771a82e26246691de6a3cd9b4e05b9/scripts/prepare_train_labels.py#L114 will ever be true.

Now, even if it gets true, we don't need to do this https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch/blob/3fced2c8fb771a82e26246691de6a3cd9b4e05b9/scripts/prepare_train_labels.py#L115. As the dictionary already contains the whole annotation including the segmentation part which was added here https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch/blob/3fced2c8fb771a82e26246691de6a3cd9b4e05b9/scripts/prepare_train_labels.py#L104.

So, I think that we can directly access the segmentation part by using `annotation['image_id']][0]['segmentation']`.

I am currently looking at COCO 2017 person keypoints json file. Thank you..

**Comment by Daniil-Osokin** [comment]
Hi, MS COCO has some `crowd` annotations - for regions, which contain the humans, but without properly labeled keypoints (due to high density of human inside these regions). To not penalize algorithm for detections in such regions, it is useful not to compute loss in them (mask it with zeros). So lines 99-104 gather properly annotated poses, lines 106-111 gather such unlabeled regions that should be masked out, and lines 113-115 combines them by image id (to have for each image proper poses and regions to mask out, useful during training data generation).

**Comment by Daniil-Osokin** [comment]
Hope, it helped.
